src/SimulationRunner.cc

```diff
@@ -319,40 +319,34 @@ void SimulationRunner::UpdateCurrentInfo()
 
   // Regular time flow
 
-  // Store the real time and sim time only if not paused.
-  if (this->realTimeWatch.Running())
-  {
-    this->realTimes.push_back(this->realTimeWatch.ElapsedRunTime());
-    this->simTimes.push_back(this->currentInfo.simTime);
-  }
-
-  // Maintain a window size of 20 for realtime and simtime.
-  if (this->realTimes.size() > 20)
-    this->realTimes.pop_front();
-  if (this->simTimes.size() > 20)
-    this->simTimes.pop_front();
+  const double simTimeCount =
+      static_cast<double>(this->currentInfo.simTime.count());
+  const double realTimeCount =
+      static_cast<double>(this->currentInfo.realTime.count());
 
-  // Compute the average sim and real times.
-  std::chrono::steady_clock::duration simAvg{0}, realAvg{0};
-  std::list<std::chrono::steady_clock::duration>::iterator simIter,
-    realIter;
-
-  simIter = ++(this->simTimes.begin());
-  realIter = ++(this->realTimes.begin());
-  while (simIter != this->simTimes.end() && realIter != this->realTimes.end())
+  if (realTimeCount > 0)
   {
-    simAvg += ((*simIter) - this->simTimes.front());
-    realAvg += ((*realIter) - this->realTimes.front());
-    ++simIter;
-    ++realIter;
-  }
-
-  // RTF, only compute this if the realTime count is greater than zero. The
-  // realtTime count could be zero if simulation was started paused.
-  if (realAvg.count() > 0)
-  {
-    this->realTimeFactor = math::precision(
-          static_cast<double>(simAvg.count()) / realAvg.count(), 4);
+    // Store the real time and sim time only if not paused.
+    constexpr double kAlpha = 0.999;
```

Comment:
I found when debugging, it's helpful to actually look at a histogram, I have used this one in the past: https://github.com/mjcarroll/ign_imgui

Maybe we should move this into a more main-line place if others would find it useful.

Reply:
Maybe we create a new `filtered_stats` topic that outputs this and the current `stats` topic can just be the instantaneous RTF (Δ simTime/Δ realTime). We can then use the `filtered_stats` for the GUI and the `stats` topic for development/debugging. For reference, gazebo-classic had an infinite window and computed `RTF = total simTime/total realTime` so it didn't fluctuate as much: https://github.com/gazebosim/gazebo-classic/blob/d3584008b47c745c1732a6cd07af68726cd900fa/gazebo/util/Diagnostics.cc#L104-L123

Reply:
> Maybe we create a new filtered_stats topic that outputs this and the current stats topic can just be the instantaneous RTF (Δ simTime/Δ realTime). 

We could also just compute the filtered stats / RTF on the GUI end instead of on the server? I haven't paid attention to the diagnostics much in classic.  As for the RTF displayed in the GUI in gazebo-classic, it's computed on the GUI side and uses a [size of 20]( https://github.com/gazebosim/gazebo-classic/blob/gazebo11/gazebo/gui/TimePanel.cc#L223-L229). It subscribes to the `~/world_stats` topic which is published at [100Hz](https://github.com/gazebosim/gazebo-classic/blob/gazebo11/gazebo/physics/World.cc#L295) (real time). So assuming sim is running at the same speed on real time, that's roughly 2 second window?

Reply:
Ah, my mistake. I assumed the RTF calculation in the diagnostics was what was displayed in the GUI of gazebo-classic. So I guess the big difference comes from the fact that it's published at 100Hz. With a size of 20, I calculate a 200ms second window, but in gz-sim, assuming we're running at 1000Hz, the window is 20ms.

My main concern was the GUI, so I can put this logic there. We can leave the RTF calculation in `SimulationRunner` intact so we don't change behavior for now, and change it to instantaneous RTF in Harmonic. Does that sound okay?

Reply:
ah yes 200ms window ha 😓 

> We can leave the RTF calculation in SimulationRunner intact so we don't change behavior for now, and change it to instantaneous RTF in Harmonic. Does that sound okay?

sounds good to me!

Reply:
Done in https://github.com/gazebosim/gz-sim/pull/1942/commits/40e7163b4f7107151f7801f0fb398e8f9f62ec1a and follow-up PR in https://github.com/gazebosim/gz-gui/pull/529

Reply:
> I found when debugging, it's helpful to actually look at a histogram, I have used this one in the past: https://github.com/mjcarroll/ign_imgui
> 
> Maybe we should move this into a more main-line place if others would find it useful.

It could be added as an example in gz-sim. Better yet, it could be a feature of the "Topic Plotting" GUI plugin 😉.